src/control/lib/hardware/sysfs/provider.go

```diff
@@ -246,63 +242,166 @@ func (s *Provider) getCXIFabricInterfaces() ([]*hardware.FabricInterface, error)
 	return cxiFIs, nil
 }
 
-// CheckFabricReady inspects the fabric interface to determine whether it is fully initialized
-// and ready for use.
-func (s *Provider) CheckFabricReady(iface string) error {
+// GetNetDevState fetches the state of a network interface.
+func (s *Provider) GetNetDevState(iface string) (hardware.NetDevState, error) {
 	if s == nil {
-		return errors.New("sysfs provider is nil")
+		return hardware.NetDevStateUnknown, errors.New("sysfs provider is nil")
 	}
 
 	if iface == "" {
-		return errors.New("fabric interface name is required")
+		return hardware.NetDevStateUnknown, errors.New("fabric interface name is required")
 	}
 
 	devClass, err := s.GetNetDevClass(iface)
 	if err != nil {
-		return errors.Wrapf(err, "can't determine device class for %q", iface)
+		return hardware.NetDevStateUnknown, errors.Wrapf(err, "can't determine device class for %q", iface)
 	}
 
-	if devClass != hardware.Infiniband {
-		// Infiniband is the only type we need to actively check
-		return nil
+	switch devClass {
+	case hardware.Infiniband:
+		return s.getInfinibandDevState(iface)
+	case hardware.Loopback:
+		return s.getLoopbackDevState(iface)
+	default:
+		return s.getNetOperState(iface)
+	}
+}
+
+func (s *Provider) getLoopbackDevState(iface string) (hardware.NetDevState, error) {
+	state, err := s.getNetOperState(iface)
+	if err != nil {
+		return hardware.NetDevStateUnknown, err
+	}
+
+	// Loopback devices with state unknown are up
+	if state == hardware.NetDevStateUnknown {
+		return hardware.NetDevStateReady, nil
+	}
+	return state, nil
+}
+
+func (s *Provider) getNetOperState(iface string) (hardware.NetDevState, error) {
+	statePath := s.sysPath("class", "net", iface, "operstate")
+	stateBytes, err := ioutil.ReadFile(statePath)
+	if err != nil {
+		return hardware.NetDevStateUnknown, errors.Wrapf(err, "failed to get %q operational state", iface)
 	}
 
+	stateStr := strings.TrimSpace(string(stateBytes))
```

Comment:
Might be worth throwing a `strings.ToLower()` on this too just for good measure.

Reply:
Good idea.